packages/preferences/src/browser/preferences-json-schema-contribution.ts

```diff
@@ -23,7 +23,7 @@ import { PreferenceConfigurations } from '@theia/core/lib/browser/preferences/pr
 import { PreferenceScope } from '@theia/core/lib/browser';
 
 const PREFERENCE_URI_PREFIX = 'vscode://schemas/settings/';
-const USER_STORAGE_PREFIX = 'user_storage:/';
+const USER_STORAGE_PREFIX = 'user-storage:/';
```

Comment:
I understand that `_` is not valid according to [RFC 3986](https://tools.ietf.org/html/rfc3986#page-17), but I'm not sure what the implications or side effects of changing the scheme are (it should most definitely be reported in the changelog).

For instance we should confirm that:
- existing user preferences, extensions, keymaps can still be loaded from the user's home
- any other potential use-cases

Reply:
@vince-fugnitto 
added remark in changelog - please see #9049 for that.
tested prefs, exts, keymaps and seems that scheme change works fine.